carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedWebElement.java

```diff
@@ -299,6 +300,42 @@ public WebElement getElement() {
         
         return this.element;
     }
+
+    private Optional<WebElement> findWebElement() {
+        Optional<WebElement> findedWebElement = Optional.empty();
+        List<WebElement> elements = searchContext.findElements(by);
+        if (!elements.isEmpty()) {
+            findedWebElement = Optional.of(elements.get(0));
+            if (elements.size() > 1) {
+                LOGGER.warn("There are more than one element found when try to refresh element {}", getName());
+            }
+        }
+        return findedWebElement;
+    }
+
+    public boolean refresh() {
+        return refresh(EXPLICIT_TIMEOUT);
+    }
+
+    public boolean refresh(long timeout) {
+        if (getBy() == null) {
+            throw new RuntimeException("'By' is not initialized in element " + name);
+        }
+
+        Optional<WebElement> foundElement = findWebElement();
+        if (foundElement.isPresent()) {
+            element = foundElement.get();
+            return true;
+        }
+        if (waitUntil(getDefaultCondition(getBy()), timeout)) {
+            element = searchContext.findElement(getBy());
+            if (element == null) {
+                throw new RuntimeException("Something went wrong when try to reinitialize element after successful waiting");
```

Comment:
add LOGGER.warn with the same message instead and return false

Reply:
resolved